### PR TITLE
GAWB-1765, GAWB-1735: passthrough tests

### PR DESCRIPTION
**Pull request by davidangb**
GAWB-1765: orch passthrough directive mangles query parameters
GAWB-1735: Add optional namespace parameter to method repo API call
plus a bunch of tests

This PR started because I wanted to write unit tests for the passthrough directives. By writing those tests, I found some bugs in the directives.

I fixed a problem where query parameters would not pass through correctly: `path?param` would get encoded and pass through as `path%3Fparam` .. which broke query parameters. This issue blocks GAWB-1735 and is mentioned in GAWB-1303.

I also fixed a problem where the `passthroughAllPaths` directive did not support a path like `foo/bar` as the source. This isn't hugely important because it doesn't block anything - and I'd like to eliminate `passthroughAllPaths` altogether (See GAWB-1303).

see individual commits to isolate the bugfixes.

There's a bunch more unit tests we could write around the passthrough directive - testing the compression flag, POST/PUT payloads, etc - but I think this is a good start.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by rushtong** [comment]
Ignore this comment. 

~Suggestion: can we have a test for html encoded segments similar to the hex-encoded test? Something like this to ensure we're not double encoding or un-encoding things:~
```
      "path with a uri-encoded segment" - {
        "should hit the path exactly" in {
          validateUri(
            "/one/%20space/three",
            "/one/%20space/three"
          )
        }
      }
```

**Comment by rushtong** [comment]
👍

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -202,61 +202,60 @@ paths:
...
+          in: query
+          description: Payload of method in WDL.
+          required: false
+          type: string
+        - name: entityType
+          in: query
+          description: Type of Method - Task or Workflow
+          required: false
+          type: string
+        - name: excludedFields
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml]: This doesn't seem to work correctly - not sure if the issue is what orchestration is sending to agora or if agora just isn't honoring this query param.

File: src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -202,61 +202,60 @@ paths:
...
-#          description: Fields to include in JSON response
-#          required: false
-#          type: string
+      description: |
+        The configurations endpoint returns all configurations the requester
+        has permissions to read that fit the filter criteria.
+      parameters:
+        - name: namespace
+          in: query
+          description: Namespace of method.
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml]: Here and throughout these parameters, this is for a configuration, not a method.

File: src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -648,58 +647,57 @@ paths:
...
+          in: query
+          description: Payload of method in WDL.
+          required: false
+          type: string
+        - name: entityType
+          in: query
+          description: Type of Method - Task or Workflow
+          required: false
+          type: string
+        - name: excludedFields
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml]: Similar issue to configurations - the response doesn't honor this query param.

File: src/main/resources/swagger/api-docs.yaml [hunk 4]
```diff
@@ -648,58 +647,57 @@ paths:
...
+          in: query
+          description: Fields to exclude in JSON response.
+          required: false
+          type: string
+        - name: includedFields
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml]: I also can't get this to work. Tried using `payload` which is returned when you get the individual method or config, but no dice. The query looks correctly formed so this is either the param orchestration is sending or a bug in agora that doesn't honor it.

**Comment by rushtong** [comment]
Confirmed in agora swagger that `excludedFields` and `includedFields` are not honored - orchestration is doing the right thing.

**Comment by davidangb** [comment]
yup, I also found that included/excluded fields don't work in agora. I've re-hidden those params in orch swagger.

**Review by rushtong — APPROVED** [review]
👍 LG!

**Pull request merged**